### PR TITLE
Update to the "Using the OpenCL backend" doxygen

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

An update of the "Using the OpenCL backend" doxygen. Now listing all currently OpenCL supported functions.

## Tests

/

## Side Effects

/

## Checklist

- [x] Copyright holder: Rok Češnovar, Univ. of Ljubljana

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: doxygen/parallelism_support/opencl_support.md [hunk 1]
```diff
@@ -94,20 +94,135 @@ Once you have done the above step, `runTests.py` should execute with the GPU ena
...
-- bernoulli_logit_glm
 - cholesky_decompose
-- categorical_logit_glm
 - gp_exp_quad_cov
 - mdivide_right_tri
 - mdivide_left_tri
 - multiplication
+- sum
+
+#### Supported distributions
```
Comment (t4c1) [inline doxygen/parallelism_support/opencl_support.md:128]: So far only lpdf/lpmf functions are supported (no cdf/lcdf/lccdf).

Comment (rok-cesnovar) [reply doxygen/parallelism_support/opencl_support.md:125]: Added a sentence that mentions that. Let me know if you would prefer to have the list explicitly mention the suffices like
```
- bernoulli_lpmf
- bernoulli_logit_lpmf
- bernoulli_logit_glm_lpmf
- beta_lpdf
...
```

Comment (t4c1) [reply doxygen/parallelism_support/opencl_support.md:125]: I expect I will start adding  cdf/lcdf/lccdf within a month or two. We will need to mention explicit suffixes at that point.

Comment (rok-cesnovar) [reply doxygen/parallelism_support/opencl_support.md:125]: Ok, explicit suffices it is! :)

File: doxygen/parallelism_support/opencl_support.md [hunk 2]
```diff
@@ -94,20 +94,128 @@ Once you have done the above step, `runTests.py` should execute with the GPU ena
...
+### Reverse mode
+
+An example of using OpenCL supported Stan Math functions for reverse mode AD is shown below: 
+
+```cpp
+using stan::math;
+
+Eigen::Matrix<double, -1, -1> A(N,M);
+Eigen::Matrix<double, -1, -1> B(M,N);
+// ... fill matrices A and B
```
Comment (SteveBronder) [inline doxygen/parallelism_support/opencl_support.md:108]: We could just replace this with something like 
```suggestion
Eigen::Matrix<double, -1, -1> A = Eigen::Matrix<double, -1, -1>::Random(N, M);
Eigen::Matrix<double, -1, -1> B = Eigen::Matrix<double, -1, -1>::Random(M, N);
```

File: doxygen/parallelism_support/opencl_support.md [hunk 3]
```diff
@@ -94,20 +94,128 @@ Once you have done the above step, `runTests.py` should execute with the GPU ena
...
+var_value<matrix_cl<double>> A_cl = to_matrix_cl(A);
+var_value<matrix_cl<double>> B_cl = to_matrix_cl(B);
+var_value<matrix_cl<double>> C_cl = A_cl * B_cl;
+var_value<Eigen::Matrix<double, -1, -1>> C = from_matrix_cl(C_cl);
+C(0,0).grad();
```
Comment (SteveBronder) [inline doxygen/parallelism_support/opencl_support.md:113]: Here we would probably want to set C's values to 1 and then run `grad()` right? @bbbales2 what was the reason we don't have `C.grad()` that fills the adjoints with 1 by default for var matrix types?

**Review by SteveBronder — APPROVED** [review]
Two little things but is fine! The second one we should probably clarify before merging as I think calling `c(0, 0).grad()` would just set the first cell's adjoint to 1 but then not the rest of the matrix. Maybe just do `stan::math::sum(C).grad()`

**Comment by SteveBronder** [comment]
Thanks for fixing these up!

**Comment by rok-cesnovar** [comment]
@SteveBronder I think we also need to regenerate the website? http://mc-stan.org/math/ is still showing 3.3.0.

File: doxygen/parallelism_support/opencl_support.md (hunk 3 above)
Comment (rok-cesnovar) [reply doxygen/parallelism_support/opencl_support.md]: Added the sum() to avoid this.

**Comment by SteveBronder** [comment]
Jenkins should be doing that when it runs on the develop branch. lemme look at this

**Comment by SteveBronder** [comment]
Huh part of it looks like it's working but it looks like there are some git failures that are happening silently when we upload the docs?

https://jenkins.mc-stan.org/blue/organizations/jenkins/Math%20Pipeline/detail/develop/946/pipeline/284

**Comment by SteveBronder** [comment]
But the `gh-pages` branch is updated

https://github.com/stan-dev/math/tree/gh-pages

I think we just need to bump the version number on the docs site because it the docs seem to have the most recent changes

**Comment by rok-cesnovar** [comment]
Ah yeah, this file also was not updated: https://github.com/stan-dev/math/blob/develop/stan/math/version.hpp
That is what threw me off. It does update otherwise yeah. Sorry for the false alarm

**Comment by rok-cesnovar** [comment]
This now also fixes all the version numbers. Should be good to merge now.

**Review by SteveBronder — APPROVED** [review]
lcbtm!

**Pull request merged**